src/datepicker/datepicker.component.ts

```diff
@@ -330,29 +330,30 @@ export class DatePicker implements OnDestroy, OnChanges, AfterViewChecked {
 			// we can either set a date value or an empty string, so we start with an empty string
 			let singleDate = "";
 			// if date is a string, parse and format
-			if (typeof dates[0] === "string") {
-				singleDate = this.flatpickrInstance.parseDate(dates[0], this.dateFormat);
+			if (typeof this.flatpickrInstance.selectedDates[0] === "string") {
```

Comment:
Ah, we use the dates from the flatpickr instance since they'll already be sorted correctly? Makes sense to me 👍